cmake/Format.cmake

```diff
@@ -14,8 +14,12 @@
 find_program(
     CLANG_FORMAT
     NAMES
-        clang-format-8 # Debian package manager, among others, provide this name
-        clang-format-mp-8.0 # MacPorts
+        clang-format-13 # Debian package manager, among others, provide this name
+        clang-format-mp-13.0 # MacPorts
+        clang-format-12 # Debian package manager, among others, provide this name
+        clang-format-mp-12.0 # MacPorts
+        clang-format-11 # Debian package manager, among others, provide this name
+        clang-format-mp-11.0 # MacPorts
```

Comment:
Do we want all of these available? Might provide weird issues where the local format and the CI format disagree.

Reply:
As far as I can tell, the results of all these version are consistent and only disagree with older version because this bug (https://reviews.llvm.org/D83564) in the older versions got fixed.
Therefore, my preference is to allow for more options (so that fewer users see a warning about an "untested" version, which is also why I removed the `<=` check and left it at a single `>` check) and revisit/update first, when we see a future breaking version (which is likely to be discovered by us first as I assume that our build/dev environments are on average more update to date than the one of typical users/supercomputer environments).

Reply:
Yeah ok that works for me.